### PR TITLE
fix: so the lineage query is less dependent on entities ids changes

### LINKED ISSUE #241: Lineage query for files inside a folder used in a renku command
When querying the `/graphql`, I can't get lineage for files inside a folder for which a lineage should be available.

I provide an example on this project: https://dev.renku.ch/projects/lorenzo.cavazzi.tech/zurich-bikes-tutorial using the following reference query:
```
query getLineage($filePath: FilePath!) {
    lineage(projectPath: "lorenzo.cavazzi.tech/zurich-bikes-tutorial", commitId: "1aabd4021acf7c7d4b669ad0d587466a0a1fc7cd", filePath: $filePath) {
        nodes {
            id
            label
        }
        edges {
            source
            target
        }
    }
}
```

**EXAMPLE**
If I query for `"filePath": "data/preprocessed/zhbikes.parquet"`, I get the following node
```
{
    "id": "/blob/91b226bce8dd0d3ae1be804e107ec7ff0f5ee0d2/data/zhbikes",
    "label": "data/zhbikes@91b226bce8dd0d3ae1be804e107ec7ff0f5ee0d2"
}
```
that is used as an edge once, so the file `data/zhbikes` must have a lineage.
***REF***: https://dev.renku.ch/projects/lorenzo.cavazzi.tech/zurich-bikes-tutorial/files/lineage/data/preprocessed/zhbikes.parquet

If I try to query for that path, or any files contained in it (example: `data/zhbikes/2019_verkehrszaehlungen_werte_fussgaenger_velo.csv`), I don't get any lineage
***REF***: https://dev.renku.ch/projects/lorenzo.cavazzi.tech/zurich-bikes-tutorial/files/lineage/data/zhbikes/2019_verkehrszaehlungen_werte_fussgaenger_velo.csv

This appears to be an error. Ideally, I would like to get the lineage in both cases (folder and children's files).

Another example is provided by SwissDataScienceCenter/renku-python#836

### PR DESCRIPTION
**Pull request by jachro**
It looks that renku-python 0.9.0 introduced some changes to the id of `Activity` and `Entity`. This fix mitigates that so the lineage query can still find some results.

As part of the PR, we are going to add an additional property to the lineage endpoint response. Each node along with `label` and `id` will also get a mandatory `type` property. The property is an enumeration which can take the following values: `ProcessRun`, `File`, `Directory`.

closes #241 
~~closes~~ addresses https://github.com/SwissDataScienceCenter/renku-python/issues/1007

### CODE REVIEW

**Comment by rokroskar** [comment]
@lorenzo-cavazzi can you approve (or request changes) here once you've tested it with your UI changes?

**Comment by lorenzo-cavazzi** [comment]
Sure. I was discussing with @jachro on a separate channel and this PR requires a fix before moving on

**Comment by lorenzo-cavazzi** [comment]
We continued the discussion offline. To recap, the idea is to add also a `path` property so that a node will look something like this

```
{
	"id": "http://lorenzo-renku-jena:3030/blob/7922ee1681d61ffe52f10bb59f28ebc7c380a3b4/src/clean_data.py",
	"label": "src/clean_data.py@7922ee1681d61ffe52f10bb59f28ebc7c380a3b4",
	"type": "File",
	"path": "src/clean_data.py"
},
{
	"id": "http://lorenzo-renku-jena:3030/blob/91b226bce8dd0d3ae1be804e107ec7ff0f5ee0d2/data/zhbikes",
	"label": "data/zhbikes@91b226bce8dd0d3ae1be804e107ec7ff0f5ee0d2",
	"type": "Directory",
	"path": "data/zhbikes"
},
{
	"id": "http://lorenzo-renku-jena:3030/commit/1aabd4021acf7c7d4b669ad0d587466a0a1fc7cd",
	"label": "renku run python src/clean_data.py data/zhbikes data/preprocessed/zhbikes.parquet",
	"type": "ProcessRun",
	"path": "<the workflow file path>"
},
```

The UI needs the `path` for the link that is added to every node, making it clickable.
In this way, we won't need anymore to extrapolate important data from the `id`.

File: acceptance-tests/src/test/scala/ch/datascience/graph/acceptancetests/knowledgegraph/LineageQuerySpec.scala [hunk 1]
```diff
@@ -97,6 +97,7 @@ class LineageQuerySpec extends FeatureSpec with GivenWhenThen with GraphServices
       lineage(projectPath: "namespace/lineage-project", filePath: "figs/grid_plot.png") {
         nodes {
           id
+          location
```
Comment (rokroskar) [inline acceptance-tests/src/test/scala/ch/datascience/graph/acceptancetests/knowledgegraph/LineageQuerySpec.scala]: you're already implementing https://github.com/SwissDataScienceCenter/renku/issues/696? :)

Comment (jachro) [reply acceptance-tests/src/test/scala/ch/datascience/graph/acceptancetests/knowledgegraph/LineageQuerySpec.scala]: :) It's just a relative path inside a project structure.

Comment (rokroskar) [reply acceptance-tests/src/test/scala/ch/datascience/graph/acceptancetests/knowledgegraph/LineageQuerySpec.scala]: I see that now, I got too excited. :)

Comment (jachro) [reply acceptance-tests/src/test/scala/ch/datascience/graph/acceptancetests/knowledgegraph/LineageQuerySpec.scala]: :)

**Pull request merged**